### PR TITLE
New: Add expand all button to scan results

### LINKED ISSUE #254: Add ability to open/close all results in the online scanner
If the site has a lot of errors it gets annoying to open/expend the result one by one.

### PR DESCRIPTION
**Pull request by ststimac**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests/#commitmessages)

## Short description of the change(s)

Fix #254

Adds expand all button each category of scan results. 

**Help wanted:** When all the results are expanded the button should say "- Close All"

### CODE REVIEW

**Comment by qzhou1607** [comment]
This is what it looks like in my machine. All the css updates are present so it shouldn't be a service worker issue. Are you sure you committed all the changes? @ststimac 

expand-all

**Comment by ststimac** [comment]
@qzhou1607 I think it may be because of the extra div I added. I wrapped

`{{#unless (isFinish ../status)}}`

in that div. That's probably why. One moment.

**Comment by ststimac** [comment]
Just making note, that @qzhou1607 and I solved the layout issue and she will be making that change.

File: src/hexo/themes/sonarwhal/source/js/scanner-common.js [hunk 1]
```diff
@@ -37,25 +37,73 @@
         item.setAttribute('aria-expanded', 'false');
     };
 
-    var toggleExpand = function (evt) {
-        var element = evt.target;
+    var updateExpandAllButton = function (element, closeAll) {
+        var expanded = closeAll;
+
+        if (typeof expanded === 'undefined') {
+            var parent = element.closest('.rule-result');
```
Comment (sarvaje) [inline src/hexo/themes/sonarwhal/source/js/scanner-common.js]: Maybe you can move this block of code to a new function `getStatus` or `someExpanded` or something similar

File: src/hexo/themes/sonarwhal/layout/partials/scan-result.hbs [hunk 2]
```diff
@@ -108,7 +108,14 @@
                         <div class="module module--primary results">
                             {{#each categories as |category|}}
                             <section class="rule-result" id="{{name}}">
-                                <h3>{{name}}</h3>
+                                <div class="rule-result--category">
+                                    <h3>{{name}}</h3>
+                                    {{#if (isFinish ../status)}}
+                                        {{#unless (noIssue category)}}
+                                        <button title="expand" class="button-expand-all closed">+ Expand All</button>
```
Comment (alrra) [inline src/hexo/themes/sonarwhal/layout/partials/scan-result.hbs]: In order to be consistent with what is used in JavaScript): `Expand` => `expand`

**Comment by qzhou1607** [comment]
@sarvaje @alrra Changes were made according to the feedback.

**Comment by molant** [comment]
@qzhou1607 can you fix the conflicts?

**Comment by qzhou1607** [comment]
@molant @alrra Conlicts are resolved.

**Pull request closed without merging**